### PR TITLE
kernel: Add key into trusted keyring

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Add key to default system keyring to support signed loadable kernel modules.

Signed-off-by: Chris Co <chrco@microsoft.com>

###### Change Log  

- Add mariner key to kernel keyring
- Rev kernel related specs

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

Local build, pipeline build

### CODE REVIEW

File: SPECS/kernel/kernel.spec [hunk 1]
```diff
@@ -15,6 +15,7 @@ Source0:        kernel-%{version}.tar.gz
 Source1:        config
 Source2:        config_aarch64
 Source3:        sha512hmac-openssl.sh
+Source4:        cbl-mariner-ca-20210127.pem
```
Comment (PawelWMS) [inline SPECS/kernel/kernel.spec:18]: I'm assuming these have to be always the same for this and the Hyper-V version of the kernel. I don't have a quick idea for a check/automation, which will make sure we're always using the same files, but I'd suggest at least adding a comment explaining that here and inside the Hyper-V's spec.

Alternatively, we could use a symbolic link and use only one "actual" file, but I'm not sure I'd like to make that precedence.

File: SPECS/kernel-hyperv/config [hunk 2]
```diff
@@ -4104,10 +4104,11 @@ CONFIG_PKCS7_MESSAGE_PARSER=y
 #
 CONFIG_MODULE_SIG_KEY="certs/signing_key.pem"
 CONFIG_SYSTEM_TRUSTED_KEYRING=y
-CONFIG_SYSTEM_TRUSTED_KEYS=""
+CONFIG_SYSTEM_TRUSTED_KEYS="certs/mariner.pem"
```
Comment (PawelWMS) [inline SPECS/kernel-hyperv/config:4107]: Do you know, if there's a way to allow appending/modifying the trusted keys in the future? I'm thinking about a scenario where we start signing with another key, while the old one is still valid. I'm also curious, if we have a strategy for revoking this key, even though I hope this scenario never happens.

File: SPECS/kernel/kernel.spec (hunk 1 above)
Comment (christopherco) [reply SPECS/kernel/kernel.spec:18]: It is not necessarily required that these certificates always be the same.

**Comment by christopherco** [comment]
https://github.com/microsoft/CBL-Mariner/pull/1000

File: SPECS/kernel-hyperv/config (hunk 2 above)
Comment (christopherco) [reply SPECS/kernel-hyperv/config:4107]: The changes also enable the .blacklist keyring so we can revoke keys if needed.

**Pull request closed without merging**